resources/idlharness.js

```diff
@@ -1514,10 +1550,10 @@ IdlInterface.prototype.test_self = function()
             // "If I was not declared with a [Constructor] extended attribute,
             // then throw a TypeError."
             assert_throws(new TypeError(), function() {
-                self[this.name]();
+                this.get_interface_object()();
             }.bind(this), "interface object didn't throw TypeError when called as a function");
             assert_throws(new TypeError(), function() {
-                new self[this.name]();
+                new this.get_interface_object()();
```

Comment:
There's a bug here. `DetectedBarcode` doesn't have a constructor in shape-detection-api.idl, but it does in Blink. A test used to fail for that but no longer does:
```diff
--- /b/s/w/ioYUXUDo/layout-test-results/external/wpt/shape-detection/idlharness.any-expected.txt
+++ /b/s/w/ioYUXUDo/layout-test-results/external/wpt/shape-detection/idlharness.any-actual.txt
@@ -7,7 +7,7 @@
 PASS FaceDetector interface: existence and properties of interface prototype object's "constructor" property
 PASS FaceDetector interface: existence and properties of interface prototype object's @@unscopables property
 PASS FaceDetector interface: operation detect(ImageBitmapSource)
-FAIL DetectedFace interface: existence and properties of interface object assert_throws: interface object didn't throw TypeError when called as a constructor function "function () { [native code] }" did not throw
+PASS DetectedFace interface: existence and properties of interface object
 PASS DetectedFace interface object length
 PASS DetectedFace interface object name
 PASS DetectedFace interface: existence and properties of interface prototype object
@@ -23,7 +23,7 @@
 PASS BarcodeDetector interface: existence and properties of interface prototype object's @@unscopables property
 FAIL BarcodeDetector interface: attribute supportedFormats assert_true: The prototype object must have a property "supportedFormats" expected true got false
 PASS BarcodeDetector interface: operation detect(ImageBitmapSource)
-FAIL DetectedBarcode interface: existence and properties of interface object assert_throws: interface object didn't throw TypeError when called as a constructor function "function () { [native code] }" did not throw
+PASS DetectedBarcode interface: existence and properties of interface object
 PASS DetectedBarcode interface object length
 PASS DetectedBarcode interface object name
 PASS DetectedBarcode interface: existence and properties of interface prototype object
```

The same has happened with the `RTCIceTransport` interface.

Unfortunately neither of those are shipped yet so you can't try against a vanilla Chrome. But could you check if perhaps the change in the condition has just made this code unreachable in this case? That would explain the incorrect pass.

Reply:
The issue was operator precedence; this fixed it:
```diff
-                new this.get_interface_object()();
+                new (this.get_interface_object())();
```

but I used a slightly different fix in the end.

(I notice that this test isn't particularly useful; it only catches a constructor without required arguments. Oh well.)